service/src/test/java/org/apache/griffin/core/job/EventServiceTest.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.griffin.core.job;
+
+import static org.apache.griffin.core.util.EntityHelper.createGriffinMeasure;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.griffin.core.exception.GriffinException;
+import org.apache.griffin.core.integration.GriffinEvent;
+import org.apache.griffin.core.integration.GriffinHook;
+import org.apache.griffin.core.integration.JobEvent;
+import org.apache.griffin.core.job.entity.BatchJob;
+import org.apache.griffin.core.job.entity.JobDataSegment;
+import org.apache.griffin.core.measure.entity.GriffinMeasure;
+import org.apache.griffin.core.measure.entity.Measure;
+import org.apache.griffin.core.util.EntityHelper;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.util.Assert;
+
+@RunWith(SpringRunner.class)
+//@TestConfiguration
+//@AutoConfigureTestEntityManager
+@DataJpaTest
+@ComponentScan("org.apache.griffin.core")
+//@SpringBootTest
+//@ContextConfiguration(classes = {JobServiceImpl.class, SchedulerFactoryBean.class, JobInstanceRepo.class})
+public class EventServiceTest {
+    @Autowired
+    private JobService jobService;
+
+    @Autowired
+    private TestEntityManager entityManager;
+
+    @Before
+    public void setup() throws Exception {
+        Class.forName("org.apache.griffin.core.integration.JobEventHook");
+        entityManager.clear();
+        entityManager.flush();
+        setEntityManager();
+    }
+
+    @Test
+    public void testAddJobEvent() throws Exception {
```

Comment:
This test is just verifying that code compiles, and when hook runs, exception is logged. Is that right?

Reply:
yes, I try to catch exception to verify. but I don't find way in spring

Reply:
`@Test(expected=UnImplementedException.class)`, or `assertThrows()` should do, what do you think?

Reply:
I think for event test, we need to mockup a event listener in test context, and assert this test listener do accept related event filed from event manager.